addons/mail/controllers/discuss/channel.py

```diff
@@ -207,3 +208,44 @@ def discuss_channel_sub_channel_fetch(self, parent_channel_id, search_term=None,
             domain.append(("name", "ilike", search_term))
         sub_channels = request.env["discuss.channel"].search(domain, order="id desc", limit=limit)
         return Store(sub_channels).add(sub_channels._get_last_messages()).get_result()
+
+    @http.route("/discuss/channel/remove_channel_member", methods=["POST"], type="jsonrpc", auth="public")
+    def discuss_channel_remove_channel_member(self, member_id):
+        channel_member = request.env["discuss.channel.member"].search([
+            ("id", "=", member_id),
+        ])
+        if not channel_member:
+            raise NotFound()
+        channel = channel_member.channel_id
+        store = Store().delete(channel_member)
+        channel_member.unlink()
+        return store.add(channel).get_result()
+
+    @http.route("/discuss/channel/set_channel_member_role", methods=["POST"], type="jsonrpc", auth="public")
+    def discuss_channel_set_channel_member_role(self, member_id, channel_role):
+        channel_member = request.env["discuss.channel.member"].search([
+            ("id", "=", member_id),
+        ])
+        if not channel_member:
+            raise NotFound()
+        # sudo: discuss.channel.member - writing channel role of a member is considered allowed
+        channel_member.sudo().write({"channel_role": channel_role})
+        return Store(channel_member).get_result()
```

Comment:
send only the changed field?
Also rather than returning by RPC (which is only working for current tab) it should send on the bus for everyone.